iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1233,6 +1281,10 @@ pub struct DirectAddrInfo {
     ///
     /// [`Endpoint::add_node_addr`]: crate::endpoint::Endpoint::add_node_addr
     pub last_alive: Option<Duration>,
+    /// Sources is a list of [`Source`] and [`Duration`]s, keeping track of all the ways we have learned about this path
+    /// We keep track of only the latest [`Duration`] for each [`Source`], keeping the size of the list of sources down to one entry per type of source.
```

Comment:
Since this is not a struct we keep track of, but generate instantaneously on call, I think this wording needs to be adjusted. For example

`Duration is calculated as the time passed from when this source was last recorded and the instant this object was created`

This should also help with clarity as requested by @flub